spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -50,32 +53,104 @@ public class GravitinoCatalog implements TableCatalog, SupportsNamespaces {
+    String gravitinoNamespace;
+    if (namespace.length == 0) {
+      gravitinoNamespace = getCatalogDefaultNamespace();
+    } else {
+      validateNamespace(namespace);
+      gravitinoNamespace = namespace[0];
+    }
+    try {
+      NameIdentifier[] identifiers =
+          gravitinoCatalogClient
+              .asTableCatalog()
+              .listTables(Namespace.of(metalakeName, catalogName, gravitinoNamespace));
+      return Arrays.stream(identifiers)
+          .map(identifier -> Identifier.of(getNamespace(identifier), identifier.name()))
+          .toArray(Identifier[]::new);
+    } catch (NoSuchSchemaException e) {
+      throw new NoSuchNamespaceException(namespace);
+    }
   }
 
   @Override
   public Table createTable(
       Identifier ident, Column[] columns, Transform[] partitions, Map<String, String> properties)
       throws TableAlreadyExistsException, NoSuchNamespaceException {
-    throw new NotSupportedException("Doesn't support creating table");
+    checkNotPathIdentifier(ident, "createTable");
+    NameIdentifier nameIdentifier =
+        NameIdentifier.of(metalakeName, catalogName, getDatabase(ident), ident.name());
+    com.datastrato.gravitino.rel.Column[] gravitinoColumns =
+        Arrays.stream(columns)
+            .map(column -> createGravitinoColumn(column))
+            .toArray(com.datastrato.gravitino.rel.Column[]::new);
+
+    Map<String, String> gravitinoProperties =
+        propertiesConverter.toGravitinoTableProperties(properties);
+    // Spark store comment in properties, we should retrieve it and pass to Gravitino explicitly.
+    String comment = gravitinoProperties.remove(ConnectorConstants.COMMENT);
+
+    try {
+      com.datastrato.gravitino.rel.Table table =
+          gravitinoCatalogClient
+              .asTableCatalog()
+              .createTable(nameIdentifier, gravitinoColumns, comment, gravitinoProperties);
+      return gravitinoAdaptor.createSparkTable(ident, table, sparkCatalog, propertiesConverter);
+    } catch (NoSuchSchemaException e) {
+      throw new NoSuchNamespaceException(ident.namespace());
+    } catch (com.datastrato.gravitino.exceptions.TableAlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
   }
 
-  // Will create a catalog specific table by invoking createSparkTable()
+  // Will create a catalog specific table
   @Override
   public Table loadTable(Identifier ident) throws NoSuchTableException {
-    throw new NotSupportedException("Doesn't support loading table");
+    checkNotPathIdentifier(ident, "loadTable");
+    try {
+      com.datastrato.gravitino.rel.Table table =
+          gravitinoCatalogClient
+              .asTableCatalog()
+              .loadTable(
+                  NameIdentifier.of(metalakeName, catalogName, getDatabase(ident), ident.name()));
+      return gravitinoAdaptor.createSparkTable(ident, table, sparkCatalog, propertiesConverter);
```

Comment:
Why do the Adaptor call createSparkTable insteaf of `loadSparkTable` here?

Reply:
`loadSparkTable` sounds like fetch a remote table to local, while `createSparkTable` just create a SparkTable object.